EventFilter/HcalRawToDigi/plugins/alpaka/HcalDigisProducerPortable.cc

```diff
@@ -0,0 +1,200 @@
+
+  private:
+    // input product tokens
+    edm::EDGetTokenT<HBHEDigiCollection> hbheDigiToken_;
+    edm::EDGetTokenT<QIE11DigiCollection> qie11DigiToken_;
+
+    // type aliases
+    using HostCollectionPhase1 = hcal::Phase1DigiHostCollection;
+    using HostCollectionPhase0 = hcal::Phase0DigiHostCollection;
+
+    using DeviceCollectionPhase1 = hcal::Phase1DigiDeviceCollection;
+    using DeviceCollectionPhase0 = hcal::Phase0DigiDeviceCollection;
+
+    // output product tokens
+    device::EDPutToken<DeviceCollectionPhase1> digisF01HEToken_;
+    device::EDPutToken<DeviceCollectionPhase0> digisF5HBToken_;
+    device::EDPutToken<DeviceCollectionPhase1> digisF3HBToken_;
+
+    struct ConfigParameters {
+      uint32_t maxChannelsF01HE, maxChannelsF5HB, maxChannelsF3HB;
+    };
+    ConfigParameters config_;
+  };
+
+  void HcalDigisProducerPortable::fillDescriptions(edm::ConfigurationDescriptions& confDesc) {
+    edm::ParameterSetDescription desc;
+
+    // FIXME
+    desc.add<edm::InputTag>("hbheDigisLabel", edm::InputTag("hcalDigis"));
+    desc.add<edm::InputTag>("qie11DigiLabel", edm::InputTag("hcalDigis"));
+    desc.add<std::string>("digisLabelF01HE", std::string{"f01HEDigisGPU"});
+    desc.add<std::string>("digisLabelF5HB", std::string{"f5HBDigisGPU"});
+    desc.add<std::string>("digisLabelF3HB", std::string{"f3HBDigisGPU"});
+    desc.add<uint32_t>("maxChannelsF01HE", 10000u);
+    desc.add<uint32_t>("maxChannelsF5HB", 10000u);
+    desc.add<uint32_t>("maxChannelsF3HB", 10000u);
+
+    confDesc.addWithDefaultLabel(desc);
+  }
+
+  HcalDigisProducerPortable::HcalDigisProducerPortable(const edm::ParameterSet& ps)
+      : hbheDigiToken_{consumes<HBHEDigiCollection>(ps.getParameter<edm::InputTag>("hbheDigisLabel"))},
+        qie11DigiToken_{consumes<QIE11DigiCollection>(ps.getParameter<edm::InputTag>("qie11DigiLabel"))},
+        digisF01HEToken_{produces(ps.getParameter<std::string>("digisLabelF01HE"))},
+        digisF5HBToken_{produces(ps.getParameter<std::string>("digisLabelF5HB"))},
+        digisF3HBToken_{produces(ps.getParameter<std::string>("digisLabelF3HB"))} {
+    config_.maxChannelsF01HE = ps.getParameter<uint32_t>("maxChannelsF01HE");
+    config_.maxChannelsF5HB = ps.getParameter<uint32_t>("maxChannelsF5HB");
+    config_.maxChannelsF3HB = ps.getParameter<uint32_t>("maxChannelsF3HB");
+  }
+
+  void HcalDigisProducerPortable::produce(device::Event& event, device::EventSetup const& setup) {
+    const auto hbheDigis = event.getHandle(hbheDigiToken_);
+    const auto qie11Digis = event.getHandle(qie11DigiToken_);
+
+    //Get the number of samples in data from the first digi
+    auto const stride = HBHEDataFrame::MAXSAMPLES * 0.5 + 1;
+    auto const size = hbheDigis->size() * stride;  // number of channels * stride
+
+    // stack host memory in the queue
+    HostCollectionPhase0 hf5_(size, event.queue());
+
+    // device product
+    DeviceCollectionPhase0 df5_(size, alpaka::getDev(event.queue()));
```

Comment:
Why allocate with device instead of queue?
```suggestion
    DeviceCollectionPhase0 df5_(size, event.queue());
```

Reply:
Done in d82fbab